applications/CompressiblePotentialFlowApplication/custom_elements/adjoint_finite_difference_potential_flow_element.h

```diff
@@ -0,0 +1,118 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//
+//  Main authors:    Marc Nuñez, based on A. Geiser, M. Fusseder, I. Lopez and R. Rossi work
+//
+
+#if !defined(KRATOS_ADJOINT_FINITE_DIFFERENCE_POTENTIAL_FLOW_ELEMENT_H_INCLUDED )
+#define KRATOS_ADJOINT_FINITE_DIFFERENCE_POTENTIAL_FLOW_ELEMENT_H_INCLUDED
+
+
+// Project includes
+#include "adjoint_base_potential_flow_element.h"
+
+namespace Kratos
+{
+
+template <class TPrimalElement>
+class AdjointFiniteDifferencePotentialFlowElement : public AdjointBasePotentialFlowElement<TPrimalElement>
+{
+public:
+
+    ///@name Type Definitions
+    ///@{
+
+    typedef AdjointBasePotentialFlowElement<TPrimalElement> BaseType;
+    static constexpr int NumNodes = TPrimalElement::TNumNodes;
```

Comment:
Both NumNodes and Dim are already defined in the base class. I think it is better not to redifine them here but instead use:

```suggestion
    using BaseType::NumNodes;
```